### PR TITLE
Regex update

### PR DESCRIPTION
**Pull request by kelvinqian00**
Address the following regex and spec-related issues:
- #70 UUID regex is outdated
- #65 Empty language maps not supported
- #64 Non-ASCII chars not supported in IRI spec
- #63 Durations spec reject valid durations
- #58 mailto spec rejects valid email usernames

The general approach was to replace monolithic regex strings with BNF-like `let` bindings, making regexes easier to debug and update.

### CODE REVIEW

File: src/xapi_schema/spec.cljc [hunk 1]
```diff
@@ -239,20 +233,39 @@
...
+        (re-matches TimestampRegEx timestamp)]
+    (cond
+      (nil? ts) ; fails regex
+      false
+      (= "02" month)
+      (let [year-int #?(:clj (Integer/parseInt year) :cljs (js/parseInt year))]
+        (if (or (and (= 0 (mod year-int 4)) (not= 0 (mod year-int 100)))
+                (= 0 (mod year-int 400)))
+          (not (#{"30" "31"} day)) ; leap year
+          (not (#{"29" "30" "31"} day))))
```
Comment (FeLungs) [inline src/xapi_schema/spec.cljc:249]: might be better to do a numeric comparison opposed to set matching.
`[day-int #?(:clj (Integer/parseInt day) :cljs (js/parseInt day))]`
....
`(not (>= 30 day-int)`
`(not (>= 29 day-int)`

^^^ method also applies to month = 4 / 6 / 9 / 11

**Review by FeLungs — COMMENTED** [review]
Really like how you broke down regexes into let binding!

Before we're done with this branch, we'll want to run it against the LRS conformance suite
- Create new Nave branch
- Update xapi-schema dep within deps.edn
- run .../script/conformance_test.sh

I believe that's it in terms of running the LRS conformance tests but lmk if you run into issues there.

**Comment by kelvinqian00** [comment]
Running the nave tests, I did run into a build error, but I doubt it has anything to do with xapi-schema:
```
Error building classpath. Failed to read artifact descriptor for com.datomic:datomic-pro:jar:0.9.5786
org.eclipse.aether.resolution.ArtifactDescriptorException: Failed to read artifact descriptor for com.datomic:datomic-pro:jar:0.9.5786
<stack trace>
```

**Comment by kelvinqian00** [comment]
After overcoming a bunch of nave setup errors, tests have succeeded:
```
Tests completed in 37.507 seconds
Test Suite Complete
Full run log written to /Users/kelvinqian/Projects/nave/lrs-conformance-test-suite/logs/9ffa3e6c-1ab3-4c91-92e1-a63ece40fceb.log
Closed
{
    "name": "console",
    "owner": null,
    "flags": {
        "endpoint": "http://localhost:9001/xapi",
        "basicAuth": true,
        "authUser": "123456789"
    },
    "options": {},
    "rollupRule": "mustPassAll",
    "uuid": "9ffa3e6c-1ab3-4c91-92e1-a63ece40fceb",
    "startTime": 1616608072650,
    "endTime": 1616608110157,
    "duration": 37507,
    "state": "finished",
    "summary": {
        "total": 1365,
        "passed": 1365,
        "failed": 0,
        "version": "1.0.3.18"
    }
}
logs/9ffa3e6c-1ab3-4c91-92e1-a63ece40fceb.log:        "failed": 0,
Conformance test exit status: 0
```

**Pull request merged**